static/app/actionCreators/group.tsx

```diff
@@ -134,7 +134,8 @@ export function assignToActor({id, actor, assignedBy}: AssignToActorParams) {
 export function deleteNote(api: Client, group: Group, id: string, _oldText: string) {
   const restore = group.activity.find(activity => activity.id === id);
   const index = GroupStore.removeActivity(group.id, id);
-  if (index === -1) {
+
+  if (index === -1 || restore === undefined) {
```

Comment:
@k-fish I had to add this check as part of the type fix, I'm not sure if it's the correct fix, but we cannot pass restore if it's undefined on L147 as the type does not allow it... Maybe it would have been better to just extend the updateActivity type here so it mimics the old code better, seems like it was supporting it anyways, wdyt?

```
promise.catch(() => GroupStore.addActivity(group.id, restore, index));
```